src/bp/ui-studio/src/web/views/OneFlow/diagram/nodes/Block/index.tsx

```diff
@@ -0,0 +1,277 @@
+import { Intent, Menu, MenuItem } from '@blueprintjs/core'
+import { FormData } from 'botpress/sdk'
+import { contextMenu, lang, ShortcutLabel } from 'botpress/shared'
+import React, { FC, useState } from 'react'
+import { AbstractNodeFactory, DiagramEngine } from 'storm-react-diagrams'
+import { BaseNodeModel } from '~/views/FlowBuilder/diagram/nodes/BaseNodeModel'
+import { StandardPortWidget } from '~/views/FlowBuilder/diagram/nodes/Ports'
+
+import ActionContents from '../ActionContents'
+import style from '../Components/style.scss'
+import NodeHeader from '../Components/NodeHeader'
+import NodeWrapper from '../Components/NodeWrapper'
+import ExecuteContents from '../ExecuteContents'
+import PromptContents from '../PromptContents'
+import RouterContents from '../RouterContents'
+import SaySomethingContents from '../SaySomethingContents'
+import TriggerContents from '../TriggerContents'
+
+interface Props {
+  node: BlockModel
+  getCurrentFlow: any
+  updateFlowNode: any
+  onDeleteSelectedElements: () => void
+  editNodeItem: (node: BlockModel, index: number) => void
+  selectedNodeItem: () => { node: BlockModel; index: number }
+  getConditions: () => any
```

Comment:
Is there a way to get the real typings dynamically, instead of using any? Like on the diagram we use something like `type DispatchProps = typeof mapDispatchToProps` to get that. Maybe type commonProps from there and pass it here ?